src/EA.Weee.Web/Areas/AATFReturn/Controllers/SentOnCreateSiteOperatorController.cs

```diff
@@ -0,0 +1,107 @@
+﻿namespace EA.Weee.Web.Areas.AatfReturn.Controllers
+{
+    using EA.Prsd.Core.Mapper;
+    using EA.Weee.Api.Client;
+    using EA.Weee.Core.AatfReturn;
+    using EA.Weee.Requests.AatfReturn.Obligated;
+    using EA.Weee.Requests.Shared;
+    using EA.Weee.Web.Areas.AatfReturn.Mappings.ToViewModel;
+    using EA.Weee.Web.Areas.AatfReturn.Requests;
+    using EA.Weee.Web.Areas.AatfReturn.ViewModels;
+    using EA.Weee.Web.Constant;
+    using EA.Weee.Web.Controllers.Base;
+    using EA.Weee.Web.Infrastructure;
+    using EA.Weee.Web.Services;
+    using EA.Weee.Web.Services.Caching;
+    using System;
+    using System.Threading.Tasks;
+    using System.Web.Mvc;
+
+    public class SentOnCreateSiteOperatorController : ExternalSiteController
+    {
+        private readonly Func<IWeeeClient> apiClient;
+        private readonly BreadcrumbService breadcrumb;
+        private readonly IWeeeCache cache;
+        private readonly IEditSentOnAatfSiteRequestCreator requestCreator;
+        private readonly IGetSentOnAatfSiteRequestCreator getRequestCreator;
+        private readonly IMap<ReturnAndAatfToSentOnCreateSiteOperatorViewModelMapTransfer, SentOnCreateSiteOperatorViewModel> mapper;
+
+        public SentOnCreateSiteOperatorController(Func<IWeeeClient> apiClient, BreadcrumbService breadcrumb, IWeeeCache cache, IEditSentOnAatfSiteRequestCreator requestCreator, IMap<ReturnAndAatfToSentOnCreateSiteOperatorViewModelMapTransfer, SentOnCreateSiteOperatorViewModel> mapper, IGetSentOnAatfSiteRequestCreator getRequestCreator)
+        {
+            this.apiClient = apiClient;
+            this.breadcrumb = breadcrumb;
+            this.cache = cache;
+            this.requestCreator = requestCreator;
+            this.getRequestCreator = getRequestCreator;
+            this.mapper = mapper;
+        }
+
+        [HttpGet]
+        public virtual async Task<ActionResult> Index(Guid returnId, Guid organisationId, Guid aatfId, Guid weeeSentOnId, bool? javascriptDisabled)
+        {
+            using (var client = apiClient())
+            {
+                var operatorCountryData = await client.SendAsync(User.GetAccessToken(), new GetCountries(false));
+
+                var viewModel = mapper.Map(new ReturnAndAatfToSentOnCreateSiteOperatorViewModelMapTransfer(operatorCountryData) { ReturnId = returnId, AatfId = aatfId, OrganisationId = organisationId, WeeeSentOnId = weeeSentOnId });
+
+                var request = getRequestCreator.ViewModelToRequest(viewModel);
+
+                var siteAddressData = await client.SendAsync(User.GetAccessToken(), request);
+
+                viewModel.SiteAddressData = siteAddressData;
+
+                if (javascriptDisabled != null)
+                {
+                    var disabled = (bool)javascriptDisabled;
```

Comment:
Do this mapping in the mapper class. Pass the javascriptDisabled action parameter in with the transfer object and do the check and setting of the operator address fields.

Or probably better... have a site address property on ReturnAndAatfToSentOnCreateSiteOperatorViewModelMapTransfer. If javascript is disabled retrieve the address and pass the address on the new property. In the mapper check that property, if its not null then set the address fields to the one passed in values.